src/Radicle/Internal/Core.hs

```diff
@@ -722,8 +735,7 @@ specialForms = Map.fromList $ first Ident <$>
       val' <- baseEval val
       case val' of
         Lambda is b e -> do
```

Comment:
Here there should also be a case for `LambdaRec` too?

Reply:
I’m not sure. I think the only way you can have a `LambdaRec` here is with the following code
~~~clojure
(def-rec foo (fn [] nil))
(def-rec bar foo)
~~~
But this does not really make

Reply:
I think it could happen quite reasonably:

```
(def-rec mock-foo ...)
(def-rec real-foo ...)

(def-rec foo (if (read-ref debug-mode?) mock-foo real-foo))
```

So I think it's fine to say that the user should use `def` rather than `def-rec` in this case, but let's then give an informative error.

Reply:
Good point. Can you have another look